### PR TITLE
Networking Architecture Module closes #468

### LINKED ISSUE #468: Networking Architecture Module
possible tasks
    +what types of network architectures are there in the building?
    +build the best network based on physical & logical topology
    +explain how youtube keeps track of counting

### PR DESCRIPTION
**Pull request by nickrstewarttds**

### CODE REVIEW

File: topics/networking/modules/architecture/README.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+```text
+  [Node] [Node]
+      \   /
+      [Hub]
+      /   \
+  [Node] [Node]
+```
+Star topology makes use of a 'middle' **Hub**, which serves to re-transmit information sent from one node to another which is intended to receive that information.
+
+As information is only sent between each node and the Hub, this theoretically stops information from spreading through the network 'in search' of the node it needs to get to.
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
As information is only sent between each node and the *Hub*, this theoretically stops information from spreading through the network 'in search' of the node it needs to get to.
```

File: topics/networking/modules/architecture/README.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+```text
+  ┌[Node]─[Node]┐
+  |    \   /    |
+  |     ───     |
+  |    /   \    |
+  └[Node]─[Node]┘
+```
+Mesh topology connects every node, where possible, to every other node in the network, without using a Hub to redirect information.
+
+By connecting every node together, a Hub becomes unnecessary, as data will be received just as quickly at any node irrespective of which node it is sent from.
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
By connecting every node together, a *Hub* becomes unnecessary, as data will be received just as quickly at any node irrespective of which node it is sent from.
```

File: topics/networking/modules/architecture/README.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+### Tree 
+```text
+          ┌────────[Router]────────┐
+       [Switch]                 [Switch]
+    ┌─────┴─────┐            ┌─────┴─────┐
+  [Node]      [Node]       [Node]      [Node]
+```
+You might recognise this setup!
+
+Tree topology essentially organises several Star-based networks into a hierarchy.
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
Tree topology essentially organises several *Star* topologies based networks into a hierarchy.
```

File: topics/networking/modules/architecture/README.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+By doing this, the 'leaves' of the Tree - the peripheral nodes - are only required to send and receive information to the node directly above it.
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
By doing this, the 'leaves' (nodes) of the Tree are only required to send and receive information to the node directly above it (switch).
```

File: topics/networking/modules/architecture/README.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+As a result, rather than have a single Hub node, there are muiltiple switches which share responsibility for re-transmitting data, and a single router to facilitate.
+
+### Hybrid P2P (Peer-to-Peer)
+As networks have expanded (and become more wireless) over time, most network architectures employ a **Hybrid** system, most commonly based on Tree architecure.
+
+Wireless technology is Hybrid by design: most Internet-enabled devices are so interconnected that both physical and logical network topology has become less relevant.
+
+When we talk about topology today, we only really refer to it when describing two things:
+1. Smaller, wired, or home/office network architecture;
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
1. **Smaller, wired, or home/office network architecture**
```

File: topics/networking/modules/architecture/README.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+2. Larger, integrated, wired networks from the pre-wireless age.
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
2. **Larger, integrated, wired networks from the pre-wireless age**
```

File: topics/networking/modules/architecture/README.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+Generally, wireless networks are interconnected, or **peer-to-peer** (P2P) networks.
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
Generally, wireless networks are interconnected, or **peer-to-peer** (**P2P**) networks.
```

File: topics/networking/modules/architecture/README.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+Every node can exchange data with any other node, with each node acting both as a client (receiving information) and a server (sending information).
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
Every node can both send and receive data.
```

File: topics/networking/modules/architecture/README.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+
+This eliminates the need for specific routers, hubs, or switches.
+
+## Tasks
+*These tasks are Team Offline Exercises. You will split into teams of 4 to come up with a solution, which you will be expected to present to the rest of the group.*
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
Try to complete the following tasks:
```

File: topics/networking/modules/architecture/README.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+### Office Network Topology
+What types of network architecture are there in the office?
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
What types of network architecture are where you currently are?
```

File: topics/networking/modules/architecture/README.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+
+We have spoken about a few different network topologies in this module.
+
+Think about the ways in which devices in this room might link into a network.
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]:
```suggestion
Think about the ways in which devices where you are might link into a network.
```

File: topics/networking/modules/architecture/README.md [hunk 12]
```diff
@@ -55,33 +62,43 @@ Depending on the network we're looking at, we may also encounter other component
...
-And the **Nodes** both send information to and receive information from the **Switches**.
+Your device needs to send information to **[Node 1]**.
 
-If one of these nodes was this computer, and it needed to send information to another node on the network, there's a lot that would need to happen:
-1. First, the **node** sends a *data packet* to the **switch** it is connected to.
-2. The **switch** receives the packet, then *forwards* it to the **router**.
-3. The **router** receives the packet, works out which **node** it needs to go to, and *forwards* it to the **switch** in charge of that node.
-4. The **switch** at the other end receives the packet, then *forwards* it down to the receiver **node**.
+To do this, there's a lot that would need to happen:
+1. First, your device at **[Node 4]** sends a *data packet* up the network to **[Switch B]**.
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]: can you remove the full stops at the end of the sentences

File: topics/networking/modules/architecture/README.md [hunk 13]
```diff
@@ -90,99 +107,109 @@ Network **topology** typically refers to the *physical* and *logical* ways in wh
...
 We have spoken about a few different network topologies in this module.
 
-Think about the ways in which devices in this room might link into a network.
+Think about the ways in which devices where you are might link into a network.
 
 You will need to consider the following:
- - Which of each of the network components in this room are connected to a wired network(s);
- - Which of the wireless devices in this room are connected to a wireless network;
- - What the overarching topology of the networks in the office as a whole might look like.
+ - Which of each of the network components where you are are connected to a wired network(s);
```
Comment (tvaidotas) [inline topics/networking/modules/architecture/README.md]: shouldn't it be question marks at the end of the sentences?

Comment (nickrstewarttds) [reply topics/networking/modules/architecture/README.md]: not in this case because they're part of a 'things to consider' list

**Pull request merged**